error: cannot convert 'struct main__AStruct' to 'struct Option_main__AStruct'

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: windows, Microsoft Windows 10 Pro v19043 64-bit
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz  

CC version: Error: exec failed (CreateProcess) with code 2: 系统找不到指定的文件。

 cmd: cc --version

getwd: C:\Users\33301\Desktop
vmodules: C:\Users\33301\.vmodules
vroot: D:\ProgrammingLanguages\v
vexe: D:\ProgrammingLanguages\v\v.exe
vexe mtime: 2022-05-05 08:13:48
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 ac90a2b.7499506

Git version: git version 2.33.0.windows.2
Git vroot status: weekly.2022.18-19-g7499506c
.git/config present: true
thirdparty/tcc status: thirdparty-windows-amd64 0137a863

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg .\prog.v`
```v
struct AStruct {
	field1 int
}

fn create() ?AStruct {
	return AStruct{123}
}

fn main() {
	dump(create())
}
```

**What did you expect to see?**

compiles fine


**What did you see instead?**
```
==================
C:/Users/33301/AppData/Local/Temp/v_0/prog.17792131759653420710.tmp.c:453: warning: WINVER redefined
C:/Users/33301/AppData/Local/Temp/v_0/prog.17792131759653420710.tmp.c:2062: error: cannot convert 'struct main__AStruct' to 'struct Option_main__AStruct'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -2670,6 +2670,7 @@ pub fn (mut c Checker) expr(node_ ast.Expr) ast.Type {
 		}
 		ast.DumpExpr {
 			node.expr_type = c.expr(node.expr)
+			c.check_expr_opt_call(node.expr, node.expr_type)
 			etidx := node.expr_type.idx()
 			if etidx == ast.void_type_idx {
 				c.error('dump expression can not be void', node.expr.pos())

--- a/vlib/v/checker/tests/optional_in_dump_err.out
+++ b/vlib/v/checker/tests/optional_in_dump_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/optional_in_dump_err.vv:10:7: error: create() returns an option, so it should have either an `or {}` block, or `?` at the end
+    8 |
+    9 | fn main() {
+   10 |     dump(create())
+      |          ~~~~~~~~
+   11 | }

--- a/vlib/v/checker/tests/optional_in_dump_err.vv
+++ b/vlib/v/checker/tests/optional_in_dump_err.vv
@@ -1,0 +1,11 @@
+struct AStruct {
+	field1 int
+}
+
+fn create() ?AStruct {
+	return AStruct{123}
+}
+
+fn main() {
+	dump(create())
+}

